gpii/node_modules/preferencesServer/src/preferencesServer.js

```diff
@@ -90,19 +94,16 @@ https://github.com/GPII/universal/blob/master/LICENSE.txt
         }
     });
 
-    gpii.preferencesServer.getRawPreferences = function (rawPreferencesDataSource, event, userToken) {
+    gpii.preferencesServer.getRawPreferences = function (rawPreferencesDataSource, errorEvent, event, userToken) {
         if (userToken === undefined) {
             fluid.fail("User token for getting preferences is undefined - aborting");
             return;
         }
 
-        rawPreferencesDataSource.get({
-            userToken: userToken
-        }, gpii.preferencesServer.logAndNotify("Fetched raw user preferences: ", event,
-            function getPreferences(data) {
-                return (data === undefined) ? {} : data.preferences || data;
-            })
-        );
+        var promise = rawPreferencesDataSource.get({userToken: userToken});
+        promise.then(function (data) {
+            event.fire((data === undefined) ? {} : data.preferences || data);
+        }, errorEvent.fire);
```

Comment:
Return this promise rather than firing the error event. Then you can eliminate the use of {requestProxy} in the binding.